common/djangoapps/third_party_auth/api/views.py

```diff
@@ -249,6 +254,49 @@ def get(self, request):
         identifier = self.get_identifier_for_requested_user(request)
         return self.do_get(request, identifier)
 
+    def delete(self, request):
+        """
+        Delete given social auth record for a user.
+
+        Args:
+            request (Request): The HTTP DELETE request
+
+        Request Parameters:
+            email/username: Must provide one of 'email' or 'username'.  If both are provided,
+            the username will be ignored.
+            uid: UID of the social auth record to delete
+
+        Return:
+            JSON serialized list of the providers linked to this user after the delete operation.
+
+        """
+        identifier = self.get_identifier_for_requested_user(request)
+        uid = request.query_params.get("uid")
+        if not uid:
+            raise exceptions.ValidationError("Must provide uid")
+
+        is_unprivileged = self.is_unprivileged_query(request, identifier)
+
+        if is_unprivileged:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        try:
+            user = User.objects.get(**{identifier.kind: identifier.value})
+            social_auth_records = UserSocialAuth.objects.filter(user=user)
+            social_auth_records.get(uid=uid).delete()
```

Comment:
I think it's better to condense this into a single query:

`UserSocialAuth.objects.get(**{"user__"+identifier.kind: identifier.value}, uid=uid).delete()`

Conventionally, delete doesn't return any data, so you can switch to a 204 status code and return an empty response.